drivers/grove/light_sensor.c

```diff
@@ -71,15 +81,22 @@ static int gls_init(struct device *dev)
 		return -EINVAL;
 	}
 
-	drv_data->sample.sampling_delay = 12;
-	drv_data->sample.channel_id = CONFIG_GROVE_LIGHT_SENSOR_ADC_CHANNEL;
-	drv_data->sample.buffer = drv_data->adc_buffer;
-	drv_data->sample.buffer_length = 4;
-
-	drv_data->adc_table.entries = &drv_data->sample;
-	drv_data->adc_table.num_entries = 1;
-
-	adc_enable(drv_data->adc);
+#if defined(CONFIG_ARC)
```

Comment:
you should only ifdef the items that are different, not the whole thing.

Reply:
For ARC board whatever was present, that's enough for now. If any extra parameters such as GAIN is required for other boards, we can add it later with if def